pydantic/networks.py

```diff
@@ -86,136 +88,294 @@ def __hash__(self) -> int:
             )
         )
 
+    @property
+    def set_constraints(self) -> dict[str, Any]:
+        """Fetch a key / value mapping of constraints to values that are not None. Used for core schema updates."""
+        return {k: v for k, v in self.__dict__.items() if v is not None}
```

Comment:
```suggestion
        return {field.name: getattr(self, field.name) for field in fields(self)}
```

using `dataclass.fields`